extensions/mssql/src/objectManagement/ui/attachDatabaseDialog.ts

```diff
@@ -0,0 +1,188 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata';
+import { ObjectManagementDialogBase, ObjectManagementDialogOptions } from './objectManagementDialogBase';
+import { DatabaseFileData, IObjectManagementService, ObjectManagement } from 'mssql';
+import { Database, DatabaseViewInfo } from '../interfaces';
+import { AttachDatabaseDocUrl } from '../constants';
+import * as loc from '../localizedConstants';
+import { RemoveText } from '../../ui/localizedConstants';
+import { DefaultMinTableRowCount, DialogButton, getTableHeight } from '../../ui/dialogBase';
+import path = require('path');
+import { getErrorMessage } from '../../utils';
+
+export class AttachDatabaseDialog extends ObjectManagementDialogBase<Database, DatabaseViewInfo> {
+	private _databasesToAttach: DatabaseFileData[] = [];
+	private _databasesTable: azdata.TableComponent;
+	private _associatedFilesTable: azdata.TableComponent;
+	private _databaseFiles: string[][] = [];
+	private readonly _fileFilters: azdata.window.FileFilters[] = [{ label: loc.DatabaseFilesFilterLabel, filters: ['*.mdf'] }];
```

Comment:
question: Attaching .mdf file is enough to attach a database? will it have the references of its related .ndf and .ldf files?

Reply:
We get all of the associated files in a separate query. This is the same file browser as SSMS.